411-dmz-example-1/azuredeploy.json

```diff
@@ -0,0 +1,521 @@
+﻿{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "newStorageAccountName": {
+            "type": "string",
+            "minLength": 3,
+            "maxLength": 24,
+            "metadata": {
+                "description": "Unique DNS Name for the Storage Account where the Virtual Machine's disks will be placed."
+            }
+        },
+        "adminUsername": {
+            "type": "string",
+            "minLength": 1,
+            "metadata": {
+                "description": "Username for the Virtual Machines."
+            }
+        },
+        "adminPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Password for the Virtual Machines."
+            }
+        }
+    },
+  "variables": {
+    "NSGName": "myVNetNSG",
+
+    "VM01Name": "IIS01",
+    "VM02Name": "AppVM01",
+    "VM03Name": "AppVM02",
+    "VM04Name": "DNS01",
+
+    "VNetName": "VNet01",
+    "Subnet1Name": "FrontEnd",
+    "Subnet1Prefix": "10.0.1.0/24",
+    "Subnet2Name": "BackEnd",
+    "Subnet2Prefix": "10.0.2.0/24",
+
+    "NIC01SubnetRef": "[concat(resourceId('Microsoft.Network/virtualNetworks', variables('VNetName')), '/subnets/', variables('Subnet1Name'))]",
+    "NIC02SubnetRef": "[concat(resourceId('Microsoft.Network/virtualNetworks', variables('VNetName')), '/subnets/', variables('Subnet2Name'))]",
+    "NIC03SubnetRef": "[concat(resourceId('Microsoft.Network/virtualNetworks', variables('VNetName')), '/subnets/', variables('Subnet2Name'))]",
+    "NIC04SubnetRef": "[concat(resourceId('Microsoft.Network/virtualNetworks', variables('VNetName')), '/subnets/', variables('Subnet2Name'))]",
+
+    "storageAccountType": "Standard_LRS",
+    "imageOffer": "WindowsServer",
+    "imagePublisher": "MicrosoftWindowsServer",
+    "windowsOSVersion": "2012-R2-Datacenter",
+    "vmSize": "Standard_D3",
+    "vmStorageAccountContainerName": "vhds"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('newStorageAccountName')]",
+      "apiVersion": "2015-05-01-preview",
```

Comment:
Use 2016-01-01 API version for storage. This is a breaking change, see 101-storage-account-create template for usage